### DISCUSSION TITLE
SwarmUI and MultiGPU nodes

### DISCUSSION THREAD

**Discussion by Blahblahdenver** [mcmonkeyprojects/SwarmUI#160] [Q&A] [answered]
I am using SwarmUI with Comfy, I've downloaded the nodes from https://github.com/neuratech-ai/ComfyUI-MultiGPU.git
I have everything setup perfectly. on windows 10. rtx4090 and 2070 installed. all GPU are visible. Cuda 12.2. when I use the dropdown menu within the comfy workflow MultiGPU nodes only Cuda 0 shows up. not Cuda 1.  Someone suggested SwarmUI is preventing both from being found.
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 536.25                 Driver Version: 536.25       CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                     TCC/WDDM  | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 4090      WDDM  | 00000000:0A:00.0  On |                  Off |
| 30%   31C    P8              33W / 450W |    429MiB / 24564MiB |      1%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce RTX 2070      WDDM  | 00000000:0B:00.0 Off |                  N/A |
|  0%   42C    P8              16W / 175W |      0MiB /  8192MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
(base) PS C:\Windows\system32> python -c "import torch; [print(torch.cuda.get_device_name(i)) for i in range(torch.cuda.device_count())]"
>>
NVIDIA GeForce RTX 4090
NVIDIA GeForce RTX 2070

**Comment by Jelosus2** [comment]
Did you read the docs about Multi GPU?
https://github.com/mcmonkeyprojects/SwarmUI/blob/master/docs/Using%20More%20GPUs.md

**Reply by Blahblahdenver** [reply]
yes, unfortunately that setting only lets you pick one.  It sets a specific GPU to use, 0 or 1, and does not let me input 0,1 to use all.

**Comment by mcmonkey4eva** [comment, marked as answer]
This should be put on the issues page, not Discussions tab - there's a conflict between Swarm's multigpu engine and that nodepack.

**Reply by Blahblahdenver** [reply]
is there a way to bypass swarmuis and just use comfyui? id like to keep swarm ui but perhaps theres a way to disable it? a toggle would be awesome.

**Reply by mcmonkey4eva** [reply]
As I said: post on the issues page, not Discussions
